File: ios/ArveyGo/ArveyGo/Utils/DashboardStrings.swift
import SwiftUI

/// Dashboard page localization strings – TR/EN/ES/FR
class DashboardStrings: ObservableObject {
    static let shared = DashboardStrings()

    @Published var currentLang: String {
        didSet { UserDefaults.standard.set(currentLang, forKey: "arveygo_lang") }
    }

    init() {
        self.currentLang = UserDefaults.standard.string(forKey: "arveygo_lang") ?? "TR"
    }

    // Navigation
    var title: String { s("Dashboard", "Dashboard", "Panel", "Tableau de bord") }
    var subtitle: String { s("Araç Takip / Genel Bakış", "Vehicle Tracking / Overview", "Rastreo / Resumen", "Suivi / Aperçu") }

    // Greeting
    var goodMorning: String { s("Günaydın", "Good Morning", "Buenos Días", "Bonjour") }
    var goodAfternoon: String { s("İyi Günler", "Good Afternoon", "Buenas Tardes", "Bon Après-midi") }
    var goodEvening: String { s("İyi Akşamlar", "Good Evening", "Buenas Noches", "Bonsoir") }
    var fleetSummaryDesc: String { s("Filonuzun güncel durumu aşağıda özetlenmiştir.", "Your fleet's current status is summarized below.", "El estado actual de su flota se resume a continuación.", "L'état actuel de votre flotte est résumé ci-dessous.") }

    // Welcome Card (Android)
    func welcomeMsg(_ name: String) -> String { s("Hoş Geldiniz, \(name)", "Welcome, \(name)", "Bienvenido, \(name)", "Bienvenue, \(name)") }
    var welcomeSubtitle: String { s("Filonuzun son durumunu görüntüleyin", "View your fleet's latest status", "Vea el estado más reciente de su flota", "Consultez le dernier état de votre flotte") }
    func kontakOnChip(_ n: Int) -> String { s("\(n) Kontak Açık", "\(n) Ignition On", "\(n) Encendido", "\(n) Contact On") }
    func kontakOffChip(_ n: Int) -> String { s("\(n) Kontak Kapalı", "\(n) Ignition Off", "\(n) Apagado", "\(n) Contact Off") }
    func bilgiYokChip(_ n: Int) -> String { s("\(n) Bilgi Yok", "\(n) No Data", "\(n) Sin Datos", "\(n) Pas de Données") }

    // Period Filter
    var fleetOverview: String { s("Filo Özeti", "Fleet Overview", "Resumen de Flota", "Aperçu de la Flotte") }
    var periodToday: String { s("Bugün", "Today", "Hoy", "Aujourd'hui") }
    var periodWeek: String { s("Hafta", "Week", "Semana", "Semaine") }
    var periodMonth: String { s("Ay", "Month", "Mes", "Mois") }
    var periodQuarter: String { s("3 Ay", "Quarter", "Trimestre", "Trimestre") }
    var periodYear: String { s("Yıl", "Year", "Año", "Année") }

    // Cards
    var vehiclesTitle: String { s("Araçlar", "Vehicles", "Vehículos", "Véhicules") }
    var viewAll: String { s("Tümünü Gör", "View All", "Ver Todo", "Voir Tout") }
    var activeVehicles: String { s("Aktif Araçlar", "Active Vehicles", "Vehículos Activos", "Véhicules Actifs") }
    var allLabel: String { s("Tümü", "All", "Todos", "Tous") }

    var driverScores: String { s("Sürücü Skorları", "Driver Scores", "Puntuaciones", "Scores Conducteurs") }
    var driverPerformance: String { s("Sürücü Performansı", "Driver Performance", "Rendimiento del Conductor", "Performance Conducteur") }
    var detailLabel: String { s("Detay", "Detail", "Detalle", "Détail") }
    var avgPrefix: String { s("Ort", "Avg", "Prom", "Moy") }

    var fleetMap: String { s("Filo Haritası", "Fleet Map", "Mapa de Flota", "Carte de Flotte") }
    var liveMapAction: String { s("Canlı Harita", "Live Map", "Mapa en Vivo", "Carte en Direct") }
    var activeLabel: String { s("Aktif", "Active", "Activo", "Actif") }
    var offlineLabel: String { s("Çevrimdışı", "Offline", "Fuera de línea", "Hors ligne") }
    var idleLabel: String { s("Rölanti", "Idle", "Ralentí", "Au Ralenti") }

    var recentAlarms: String { s("Son Alarmlar", "Recent Alarms", "Alarmas Recientes", "Alarmes Récentes") }

    var aiInsights: String { s("AI Filo İçgörüleri", "AI Fleet Insights", "Insights AI de Flota", "Insights IA de Flotte") }
    var aiAnalysis: String { s("AI Filo Analizi", "AI Fleet Analysis", "Análisis AI de Flota", "Analyse IA de Flotte") }

    // AI insights
    func aiSummary(online: Int, km: String) -> String {
        s("Filonuzda **\(online)** araç aktif durumda. Günlük toplam **\(km) km** yol katedildi.",
          "**\(online)** vehicles active in your fleet. Total **\(km) km** covered today.",
          "**\(online)** vehículos activos en su flota. Total **\(km) km** recorridos hoy.",
          "**\(online)** véhicules actifs dans votre flotte. Total de **\(km) km** parcourus aujourd'hui.")
    }
    var highPriority: String { s("Yüksek", "High", "Alta", "Haute") }
    var lowPriority: String { s("Düşük", "Low", "Baja", "Basse") }

    // Settings page
    var settingsTitle: String { s("Ayarlar", "Settings", "Configuración", "Paramètres") }
    var languageLabel: String { s("Dil", "Language", "Idioma", "Langue") }
    var appInfoTitle: String { s("Uygulama Bilgisi", "App Info", "Info de la App", "Info de l'App") }

    // ---------- Helper ----------
    private func s(_ tr: String, _ en: String, _ es: String, _ fr: String) -> String {
        switch currentLang {
        case "EN": return en
        case "ES": return es
        case "FR": return fr
        default: return tr
        }
    }
}

File: android/app/src/main/java/com/arveya/arveygo/utils/DashboardStrings.kt
package com.arveya.arveygo.utils

import android.content.Context
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

/**
 * Dashboard page localization strings – TR/EN/ES/FR
 */
object DashboardStrings {
    private val _currentLang = MutableStateFlow("TR")
    val currentLang: StateFlow<String> = _currentLang

    private var appContext: Context? = null

    fun initialize(context: Context) {
        appContext = context.applicationContext
        val saved = context.getSharedPreferences("arveygo_prefs", Context.MODE_PRIVATE)
            .getString("app_lang", "TR") ?: "TR"
        _currentLang.value = saved
    }

    fun setLanguage(lang: String) {
        _currentLang.value = lang
        appContext?.getSharedPreferences("arveygo_prefs", Context.MODE_PRIVATE)
            ?.edit()?.putString("app_lang", lang)?.apply()
    }

    // Navigation
    val title: String get() = s("Dashboard", "Dashboard", "Panel", "Tableau de bord")
    val subtitle: String get() = s("Filo Yönetim Paneli", "Fleet Management Panel", "Panel de Gestión de Flota", "Panneau de Gestion de Flotte")

    // Welcome
    fun welcomeMsg(name: String): String = s("Hoş Geldiniz, $name", "Welcome, $name", "Bienvenido, $name", "Bienvenue, $name")
    val welcomeSubtitle: String get() = s("Filonuzun son durumunu görüntüleyin", "View your fleet's latest status", "Vea el estado más reciente de su flota", "Consultez le dernier état de votre flotte")
    fun kontakOnChip(n: Int): String = s("$n Kontak Açık", "$n Ignition On", "$n Encendido", "$n Contact On")
    fun kontakOffChip(n: Int): String = s("$n Kontak Kapalı", "$n Ignition Off", "$n Apagado", "$n Contact Off")
    fun bilgiYokChip(n: Int): String = s("$n Bilgi Yok", "$n No Data", "$n Sin Datos", "$n Pas de Données")

    // Period
    val periodToday: String get() = s("Bugün", "Today", "Hoy", "Aujourd'hui")
    val periodWeek: String get() = s("Hafta", "Week", "Semana", "Semaine")
    val periodMonth: String get() = s("Ay", "Month", "Mes", "Mois")
    val periodYear: String get() = s("Yıl", "Year", "Año", "Année")

    // Cards
    val activeVehicles: String get() = s("Aktif Araçlar", "Active Vehicles", "Vehículos Activos", "Véhicules Actifs")
    val allLabel: String get() = s("Tümü", "All", "Todos", "Tous")
    val driverPerformance: String get() = s("Sürücü Performansı", "Driver Performance", "Rendimiento del Conductor", "Performance Conducteur")
    val detailLabel: String get() = s("Detay", "Detail", "Detalle", "Détail")
    val recentAlarms: String get() = s("Son Alarmlar", "Recent Alarms", "Alarmas Recientes", "Alarmes Récentes")
    val aiInsights: String get() = s("AI Filo İçgörüleri", "AI Fleet Insights", "Insights AI de Flota", "Insights IA de Flotte")

    // Settings
    val settingsTitle: String get() = s("Ayarlar", "Settings", "Configuración", "Paramètres")
    val languageLabel: String get() = s("Dil", "Language", "Idioma", "Langue")
    val appInfoTitle: String get() = s("Uygulama Bilgisi", "App Info", "Info de la App", "Info de l'App")
    val appInfoApp: String get() = s("Uygulama", "Application", "Aplicación", "Application")
    val appInfoPlatform: String get() = s("Platform", "Platform", "Plataforma", "Plateforme")
    val appInfoDeveloper: String get() = s("Geliştirici", "Developer", "Desarrollador", "Développeur")
    val legalTitle: String get() = s("Yasal", "Legal", "Legal", "Légal")
    val termsOfUse: String get() = s("Kullanım Koşulları", "Terms of Use", "Términos de Uso", "Conditions d'utilisation")
    val privacyPolicy: String get() = s("Gizlilik Politikası", "Privacy Policy", "Política de Privacidad", "Politique de confidentialité")
    val allRightsReserved: String get() = s("Tüm hakları saklıdır.", "All rights reserved.", "Todos los derechos reservados.", "Tous droits réservés.")
    val notificationSettings: String get() = s("Bildirim Ayarları", "Notification Settings", "Configuración de Notificaciones", "Paramètres de notifications")
    val notificationSettingsSubtitle: String get() = s("Push, kategoriler, sessiz saatler", "Push, categories, quiet hours", "Push, categorías, horas silenciosas", "Push, catégories, heures silencieuses")

    // Sidebar menu sections
    val menuSectionMain: String get() = s("ANA MENÜ", "MAIN MENU", "MENÚ PRINCIPAL", "MENU PRINCIPAL")
    val menuSectionFleet: String get() = s("FİLO YÖNETİMİ", "FLEET MANAGEMENT", "GESTIÓN DE FLOTA", "GESTION DE FLOTTE")
    val menuSectionMonitor: String get() = s("İZLEME", "MONITORING", "MONITOREO", "SURVEILLANCE")
    val menuSectionSettings: String get() = s("AYARLAR", "SETTINGS", "CONFIGURACIÓN", "PARAMÈTRES")
    val menuSectionSupport: String get() = s("DESTEK", "SUPPORT", "SOPORTE", "ASSISTANCE")

    // Sidebar menu items
    val menuDashboard: String get() = s("Dashboard", "Dashboard", "Panel", "Tableau de bord")
    val menuLiveMap: String get() = s("Canlı Harita", "Live Map", "Mapa en Vivo", "Carte en Direct")
    val menuRouteHistory: String get() = s("Rota Geçmişi", "Route History", "Historial de Rutas", "Historique des Trajets")
    val menuVehicles: String get() = s("Araçlar", "Vehicles", "Vehículos", "Véhicules")
    val menuDrivers: String get() = s("Sürücüler", "Drivers", "Conductores", "Conducteurs")
    val menuMaintenance: String get() = s("Bakım / Belgeler / Masraflar", "Maintenance / Docs / Expenses", "Mantenimiento / Docs / Gastos", "Maintenance / Docs / Dépenses")
    val menuAlarms: String get() = s("Alarmlar", "Alarms", "Alarmas", "Alarmes")
    val menuGeofence: String get() = s("Geofence", "Geofence", "Geocerca", "Géofence")
    val menuReports: String get() = s("Raporlar", "Reports", "Informes", "Rapports")
    val menuSettings: String get() = s("Ayarlar", "Settings", "Configuración", "Paramètres")
    val menuSupport: String get() = s("Destek Talebi", "Support Request", "Solicitud de Soporte", "Demande d'assistance")
    val menuLogout: String get() = s("Çıkış Yap", "Log Out", "Cerrar Sesión", "Se déconnecter")
    val menuCompany: String get() = s("Arveya Teknoloji", "Arveya Technology", "Arveya Tecnología", "Arveya Technologie")

    fun t(tr: String, en: String, es: String, fr: String): String = s(tr, en, es, fr)

    // ---------- Helper ----------
    private fun s(tr: String, en: String, es: String, fr: String): String =
        when (_currentLang.value) {
            "EN" -> en
            "ES" -> es
            "FR" -> fr
            else -> tr
        }
}
